[Spring 체스 - 2단계] 정(정진우) 미션 제출합니다.

bugoverdose opened the pull request.

bugoverdose committed:
refactor(board): change BoardMapGeneratorUtil to BoardFactory

- move package from util to board
- cleanse map init logic
- return initialized Board instead of hashMap

bugoverdose committed:
feat(controller): cleanse url, http methods, handler method names

bugoverdose committed:
refactor(gradle): remove unused spark dependencies

- remove unused file

bugoverdose committed:
docs(README): setup feature list

bugoverdose committed:
test(study): check hashing with salt

- cf. SecretKeyFactory, PBEKeySpec

bugoverdose committed:
feat(auth): implement basic password hashing with salt

bugoverdose committed:
feat(dao): setup new GameDao2 with name and hash columns

- add minimum test cases
- setup new game2 table

bugoverdose committed:
feat(dao): implement GameDao2 and GameEntity

- implement findAll, deleteGame
- implement all exception cases
- add test cases

bugoverdose committed:
test(fixture): implement GameDao2Stub

bugoverdose committed:
feat(service): implement ChessService2

- add test cases

bugoverdose committed:
feat(service): implement findGames method and responseDto

- add test case

bugoverdose committed:
feat(web): setup html, css for /games page

bugoverdose committed:
feat(web): setup html, css for /game page

- change init game logic at client side
- move to /game on clicking init game button

bugoverdose committed:
feat(controller): replace legacy service with new service bean

bugoverdose committed:
feat(web): remove search page and cleanse ui

bugoverdose committed:
docs(README): update feature list

bugoverdose committed:
refactor(chess): remove and replace legacy dao and service

bugoverdose committed:
refactor(dao): replace legacy game table with game2

bugoverdose committed:
refactor(dto): cleanse dto names

bugoverdose committed:
refactor(web): cleanse template paths

bugoverdose committed:
docs(README): update feature list

bugoverdose committed:
feat(controller): handle search as get method instead of post

bugoverdose committed:
refactor(css): form with single input to reusable module

bugoverdose committed:
feat(chess): render game name info at "/game/:id"

bugoverdose committed:
feat(chess): render game name info at "/result/:id"

bugoverdose committed:
feat(css): cleanse css for form styling

bugoverdose committed:
feat(html): update title contents

bugoverdose committed:
feat(EventDao): implement deleteAllByGameId method

- add test cases
- update test fixture

bugoverdose committed:
feat(chess): implement delete game request handler

bugoverdose committed:
feat(web): implement delete game functionality on client side

bugoverdose committed:
refactor(chess): cleanse controllers and dtos

bugoverdose committed:
refactor(dto): cleanse dto package structure

- separate view and json response dto

bugoverdose committed:
refactor(event): remove getters with toEntity

bugoverdose committed:
refactor(event): integrate typeChecking methods

bugoverdose committed:
feat(dao): implement Command and CommandResult

- cleanse exception handling for updates

bugoverdose committed:
feat(dao): implement Query for exception handling

bugoverdose committed:
refactor(dao): cleanse exception handling logic

bugoverdose committed:
docs(README): update feature list

bugoverdose commented on src/main/java/chess/domain/auth/AuthCredentials.java:
```diff
@@ -0,0 +1,56 @@
+
+public class AuthCredentials {
+
+    private static final String HASH_ALGORITHM = "PBKDF2WithHmacSHA1";
+    private static final byte[] SECRET_SALT = "should_be_hidden_before_deployment".getBytes();
+    private static final int HASH_ITERATION_COUNT = 65536;
+    private static final int KEY_BIT_LENGTH = 128;
+    private static final SecretKeyFactory hashFactory;
```
비밀번호의 해쉬값을 DB에 저장해봤는데요. 혹시 위의 SECRET_SALT와 같은 값을 만일 시스템 환경변수로 사용하려면 어떻게 관리할 수 있을까요?
application.yml에서 환경변수를 읽어들이는 작업은 해봤고, @Value로 스프링 빈에 값을 넣을 수 있다고 들어는 봤는데,
지금처럼 **임의의 자바 코드 내로 환경변수를 가져오는 작업**은 보통 어떻게 하는지 혹시 의견 들어볼 수 있을까요?

bugoverdose commented on src/main/java/chess/service/ChessService.java:
```diff
@@ -28,48 +32,63 @@ public ChessService(GameDao gameDao, EventDao eventDao) {
-        Game game = currentSnapShotOf(gameId);
-        validateGameOver(game);
-        return new GameResultDto(gameId, game);
+    @Transactional
+    public void deleteFinishedGame(int gameId, AuthCredentials authCredentials) {
+        gameDao.deleteGame(authCredentials.toEncrypted());
+        eventDao.deleteAllByGameId(gameId);
     }
```
이런 서비스 로직의 경우 각 Dao들만 제대로 테스트되었다면 굳이 별도로 테스트할 필요가 없다고 느껴지는데, 혹시 의견 들어볼 수 있을까요? 

사실 현재 컨트롤러를 테스트하지 않은 이유도 **내부 로직 측면에서 이미 충분히 테스트되었다**고 느껴서인데, 지금과 같은 상황에서 컨트롤러를 테스트해야할 필요성이 있을까요?

bugoverdose commented on src/main/java/chess/service/ChessService.java:
```diff
@@ -28,48 +32,63 @@ public ChessService(GameDao gameDao, EventDao eventDao) {
-        Game game = currentSnapShotOf(gameId);
-        validateGameOver(game);
-        return new GameResultDto(gameId, game);
+    @Transactional
+    public void deleteFinishedGame(int gameId, AuthCredentials authCredentials) {
+        gameDao.deleteGame(authCredentials.toEncrypted());
+        eventDao.deleteAllByGameId(gameId);
     }
```
이런 서비스 로직의 경우 각 Dao들만 제대로 테스트되었다면 굳이 별도로 테스트할 필요가 없다고 느껴지는데, 혹시 의견 들어볼 수 있을까요?

사실 현재 컨트롤러를 테스트하지 않은 이유도 내부 로직 측면에서 이미 충분히 테스트되었다고 느껴서인데, 지금과 같은 상황에서 컨트롤러를 테스트해야할 필요성이 있을까요?

JunHoPark93 replied:
두 가지 질문인거죠?
개인적으로 저는 서비스 테스트는 하고있어요. 흐름관리 측면에서 필요하다고 생각해요. 위 코드에 예시를 들자면 어떤 경우에 dao가 불리면 안되는경우에 -> 호출이 되면 안되는 테스트를 하는것이죠.
컨트롤러는 저는 안합니다. 특별한 로직이 없기도하고 투자비용 대비 얻는게 별로 없다고 생각해요.

위에는 제가 하는방식이고 더 좋은 방법이 있다면 그렇게 하셔도됩니다!

bugoverdose commented on src/main/java/chess/domain/auth/AuthCredentials.java:
```diff
@@ -0,0 +1,56 @@
+
+public class AuthCredentials {
+
+    private static final String HASH_ALGORITHM = "PBKDF2WithHmacSHA1";
+    private static final byte[] SECRET_SALT = "should_be_hidden_before_deployment".getBytes();
+    private static final int HASH_ITERATION_COUNT = 65536;
+    private static final int KEY_BIT_LENGTH = 128;
+    private static final SecretKeyFactory hashFactory;
```
비밀번호의 해쉬값을 DB에 저장해봤는데요. 혹시 위의 SECRET_SALT와 같은 값을 만일 시스템 환경변수로 사용하려면 어떻게 관리할 수 있을까요?
application.yml에서 환경변수를 읽어들이는 작업은 해봤고, `@Value`로 스프링 빈에 값을 넣을 수 있다고 들어는 봤는데,
지금처럼 **임의의 자바 코드 내로 환경변수를 가져오는 작업**은 보통 어떻게 하는지 (아니면 애초에 하지 않는지) 혹시 의견 들어볼 수 있을까요?

bugoverdose replied:
생각해보니 제대로 할거면 salt를 사용자별로 생성해서 DB에 저장해줘야 하나 싶네요

JunHoPark93 replied:
- @Value에 디폴트로 하드코딩한다. or 말씀주신 yml
- 배포 시 환경변수가 주입되도록 한다. 
- configuration server내의 값을 보도록한다. (실시간업데이트도 가능)

제가 사용했던 방식은 위와 같은데요, 두 번째 것은 상용배포플랫폼에서 거의 다 지원할거에요. 추후에 프로젝트 하실때 이런 방식을 사용하셔도 좋습니다. 저는 실무에서 1, 3번방식을 사용하고있습니다.

JunHoPark93 replied:
넵 그렇게 해주셔도되고 실제로 패스워드를 직접 관리하는 방식은 프로젝트를 진행하실때 오래 고민해보셔도 좋을거에요~ 지금은 이정도로 충분합니다.

bugoverdose commented on src/main/java/chess/dao/Command.java:
```diff
@@ -0,0 +1,32 @@
+        this.command = command;
+    }
+
+    public void executeOrThrow(String exceptionMessage) {
+        try {
+            command.getAsInt();
+        } catch (DataAccessException e) {
+            throw new IllegalArgumentException(exceptionMessage);
+        }
+    }
```
sql문의 실행 과정에서 예외가 발생할 수 있는 경우는 try-catch문으로 감싸는 방법밖에 안보여서 이렇게 Command, Query, CommandResult와 같은 클래스들을 만들었는데 이게 맞는 방법인지 잘 모르겠습니다.

JunHoPark93 replied:
하단에 남겨두었어요!

bugoverdose commented on docker/db/mysql/init/init.sql:
```diff
@@ -1,8 +1,11 @@
 CREATE TABLE game
 (
     id    BIGINT NOT NULL AUTO_INCREMENT,
+    name VARCHAR(20) NOT NULL,
+    password VARCHAR(100) NOT NULL,
```
비밀번호를 저장하라길래 hash(+salt)해서 저장해봤습니다. 해쉬값(byte[])을 그대로 DB에 Blob 타입으로 저장하려다가 테스트할 방법을 모르겠어서 다시 String으로 변환한 후 varchar 타입으로 저장했는데 이게 맞나 싶습니다.

JunHoPark93 replied:
잘하셨어요~ 제가 하는 프로젝트도 실제 위와 유사한 방법으로 하고있는데요 (salt는 쓰지 않았습니다) 테스트할 방법을 모르겠다는게 정확히 어떤 말인가요?

bugoverdose replied:
아 넵. 현재 `schema.sql`에서 `CREATE TABLE`, `INSERT`문을 활용하여 DB에 데이터를 저장해놓고, 이 저장된 데이터를 활용하도록 테스트를 작성했는데요.

`INSERT`문으로 특정한 BLOB 데이터 값을 저장하는 방법을 모르겠어서 문제가 생겼어요.
blob과는 달리 varchar 값은 INSERT문에서 다루기 쉬운 것 같아서 byte[]를 다시 String으로 변환해서 저장했다는 의미였어요.

JunHoPark93 replied:
아 그말씀이었군요, 찾아보니 뭔가 방법은 있는거같긴하네요.
저도 해당타입으로 운영해본적이 없어서 잘 모르겠지만, sql문 초기세팅이 잘 안된다고한다면 테스트코드 내에 setup으로 넣는것도 방법이긴 하겠네요.

bugoverdose commented on src/main/java/chess/controller/GameController.java:
```diff
@@ -26,29 +29,40 @@ public GameController(ChessService chessService) {
+        return modelAndView;
+    }
+
+    @PostMapping
+    public CreatedGameDto createGame(@RequestBody AuthCredentials authCredentials) {
+        return chessService.initGame(authCredentials);
     }
```
`/game/new`대신 `/game`에 대해 Post 메서드 요청을 날리는게 더 REST API스러운 것 같아 수정했습니다.

JunHoPark93 replied:
지금 단계에서는 굳이 남길필요가 없어서 남기지 않았는데 잘 수정해주셨네요~

bugoverdose commented on src/main/java/chess/controller/GameController.java:
```diff
@@ -26,29 +29,40 @@ public GameController(ChessService chessService) {
-        modelAndView.addObject(RESPONSE_MODEL_KEY, gameDto);
-        return modelAndView;
+    public ModelAndView findAndRenderGame(@PathVariable int id) {
+        return getGameModelAndView(id);
     }
 
-    @PostMapping("/{id}")
-    private ModelAndView playGame(@PathVariable int id,
-                                  @RequestBody MoveRoute moveRoute) {
+    @PutMapping("/{id}")
+    public ModelAndView updateGame(@PathVariable int id,
+                                   @RequestBody MoveRoute moveRoute) {
         chessService.playGame(id, new MoveEvent(moveRoute));
+        return getGameModelAndView(id);
+    }
```
게임을 플레이하는 과정의 경우 DB에 event가 저장되기 때문에 별 생각 없이 POST라고 했었는데, `game/:id`에 해당되는 자원이 수정된다고 보는게 더 적합하지 않나?라는 생각이 들어 PUT 메서드로 수정했습니다.

bugoverdose commented on src/main/java/chess/controller/SearchController.java:
```diff
@@ -1,38 +1,23 @@
-        return modelAndView;
-    }
-
-    @PostMapping
     public SearchResultDto searchResult(@RequestParam(name = "game_id") int gameId) {
         return chessService.searchGame(gameId);
     }
```
/search 페이지는 없앴습니다. 검색 요청 기능은 더 이상 필요 없기는 하지만 새로 만든 `/games` 페이지의 일부분으로 남겨놓았습니다.

검색 요청의 경우 POST보다는 GET 요청이 맞는 것 같아 수정해봤고, URL의 경우 이미 `game/:id`에 대한 GET 요청이 별도로 존재하기 때문에 지금처럼 `/search?game_id=XXX` 형식으로 요청을 받도록 했습니다.

JunHoPark93 replied:
https://github.com/woowacourse/jwp-chess/pull/376#discussion_r857112261
위와 같이 남겨주셨기에 남깁니다. search라는 단어가 꼭 필요할까요?

bugoverdose replied:
`SearchController`의 존재가 이상하다는 생각이 들기는 하네요.
그런데 이미 `/game`과 `/game/:id`에 대한 GET 요청이 별도로 존재해서, 라우트가 중복되지 않도록 검색 기능은 `/game/search`처럼 바꾸어야 하나 고민되네요.

JunHoPark93 replied:
음 예를 들어, /games/{gameId}/info 이렇게 표현해도 될거같아요. 리소스를 달라고하는건 이미 GET에서 드러나니까 search라는 동사는 필요없다고 생각해요.

bugoverdose replied:
아 그렇네요. 해당 데이터를 GET하는거니까 명사로 해야하네요. 제 머리가 굳어있던거 같아요

bugoverdose reviewed (commented):
안녕하세요 정입니다. 
이미 구현해놓은 것들이 많아서 생각보다 빨리 끝나버렸습니다. 피드백 바랍니다. 감사합니다:)

1. 기존 프로젝트 구조를 최소한만 건들이는 선에서 요구사항을 충족하고자 해봤습니다. 
2. domain, entity, dto로 패키지들을 분리하여 game과 event 객체들을 맥락에 따라 구분하고 있는데 오히려 전체 구조가 복잡해진건 아닌가하는 생각이 드네요. 특히 DTO들의 관리가 너무 복잡한데 view 관련 기능 때문에 필연적인 문제인지 아니면 개선할 수 있는 사항인지 고민됩니다. 
3. 컨트롤러에 대한 테스트는 필요한건지 의문이 듭니다.

JunHoPark93 commented on src/main/java/chess/domain/auth/AuthCredentials.java:
```diff
@@ -0,0 +1,56 @@
+import javax.crypto.SecretKeyFactory;
+import javax.crypto.spec.PBEKeySpec;
+
+public class AuthCredentials {
```
게임을 시작하고 방을 만들면 여기서 에러가 나는데요, 확인 한번 부탁드려요~
<img width="658" alt="스크린샷 2022-04-25 오후 8 34 13" src="https://user-images.githubusercontent.com/20608121/165081128-f1049ecd-4eda-47e4-b1f2-d93af0df953f.png">

bugoverdose replied:
혹시 3306 포트에 다른 mysql 서버가 동작하고 있는지 체크해주실 수 있나요?

만일 별도의 DB가 실행되고 있지 않다면 이 흐름대로 한번만 더 실행해주실 수 있나요.
1. 도커 데스크탑이 켜진 상태에서, 
2. 프로젝트의 docker 디렉토리에서 `docker-compose -p 컨테이너명 up -d`를 실행한 후,
3. 정상적으로 컨테이너가 생성됐을 때 애플리케이션 실행

JunHoPark93 replied:
Cannot construct instance of `chess.domain.event.MoveRoute` (no Creators,like default constructor, exist) 
기물 움직일때도 발생하고있어요

JunHoPark93 replied:
음 밑에 영상올려주신것보니 정상동작하는모습이군요

JunHoPark93 replied:
동작 하는것 확인했어요~

JunHoPark93 commented on src/main/java/chess/service/ChessService.java:
```diff
@@ -28,48 +32,63 @@ public ChessService(GameDao gameDao, EventDao eventDao) {
+
+    @Transactional
+    public CreatedGameDto initGame(AuthCredentials authCredentials) {
+        int gameId = gameDao.saveAndGetGeneratedId(authCredentials.toEncrypted());
```
AuthCredentials가 서비스레이어까지 관통하고있는데요, controller단에서 toEncrypted()로 넘겨주는건 어떨까요?

bugoverdose replied:
넵 그런 방식으로도 수정해봤고([09cd8ff](https://github.com/woowacourse/jwp-chess/pull/376/commits/09cd8ffa99c595923da6f70556bc92b31e3b8e12)),
아예 `HandlerMethodArgumentResolver` 구현체에서 toEncrypted()를 실행하고 컨트롤러에 `EncryptedAuthCredentials` 데이터가 오도록 수정하는 것도 한번 해봤어요([b4fa8ef](https://github.com/woowacourse/jwp-chess/pull/376/commits/b4fa8ef67f037a359c6eab87fd1e8bb91951dd89))

혹시 response body에 담긴 JSON을 활용하는 작업을 하고 싶으면 이렇게 하는게 맞을려나요?

JunHoPark93 replied:
argumentResolver도 사용해보셨네요!
넵 그렇게 하셔도됩니다. 처음에 드린 의도는 클라이언트에 종속된 값들중에서 필요한 값들만 서비스레이어에 전달하면 좋겠다의 의미였는데 지금구조도 괜찮습니다.

JunHoPark93 commented on src/main/java/chess/dao/Command.java:
```diff
@@ -0,0 +1,32 @@
+        this.command = command;
+    }
+
+    public void executeOrThrow(String exceptionMessage) {
+        try {
+            command.getAsInt();
+        } catch (DataAccessException e) {
+            throw new IllegalArgumentException(exceptionMessage);
```
이 구조에서 얻는 이점이 어떤것이라고 생각하셨나요?
int를 반환하게 하는 메서드를 실행하는데 DataAccessException 을 캐치하는게 어색하다고 생각해요.

쿼리를 하는 구조를 이렇게 가져간다고하면 Command를 좀 더 명시적으로 데이터베이스를 다루는 클래스임을 나타내는게 더 좋을거같은데 어떻게 생각하시나요?

bugoverdose replied:
Dao의 코드를 깔끔하게 한다는 목적이었는데, IntSupplier를 사용하는 것도 그렇고 전반적으로 내부 구조가 이상하다는 생각은 확실히 드네요.

bugoverdose replied:
흠. 애초에 의도적으로 상황에 따라 실행 실패하도록 SQL문을 작성해놓은게 잘못된건가 하는 생각도 드네요.

JunHoPark93 replied:
한곳에 몰아두셨군요. 개인적으로 StatementExecutor 보다는 DatabaseStatementExecutor같이 정이 만들어주신 의도를 좀 더 나타내면 그래도 내부 구조에 대한 설명이 좀 되지 않을까 하는 이야기였어요~

JunHoPark93 commented on src/main/java/chess/dao/CommandResult.java:
```diff
@@ -0,0 +1,16 @@
+        this.effectedRowCount = effectedRowCount;
+    }
+
+    public void throwOnNonEffected(String exceptionMessage) {
+        if (effectedRowCount == 0) {
+            throw new IllegalArgumentException(exceptionMessage);
+        }
+    }
```
위에 남긴 코멘트와 비슷한 부분입니다.
여기서는 반영된 row수가 하나라도 없다면 예외를 발생시키고있는데요, 여기서는 DataAccessException을 안다루어줘도 되는걸까요?

bugoverdose replied:
3가지 클래스를 `StatementExecutor<T>`라는 클래스 1개로 통합해보았습니다

JunHoPark93 commented on src/main/java/chess/service/ChessService.java:
```diff
@@ -28,48 +32,63 @@ public ChessService(GameDao gameDao, EventDao eventDao) {
+        Game game = currentSnapShotOf(gameId);
+        validateGameOver(game);
+        FullGameDto fullGame = new FullGameDto(gameEntity.toDto(), game.toDtoOf(gameId));
+        return new GameResultDto(fullGame, game.getResult());
```
> 특히 DTO들의 관리가 너무 복잡한데 view 관련 기능 때문에 필연적인 문제인지 아니면 개선할 수 있는 사항인지 고민됩니다.

체스 정보들이 복잡하기 때문에 화면에 그려줘야할 데이터도 복잡하게 질수밖에 없는데요, 
dto자체의 복잡성을 의미하셨다면 화면에서 꼭 다 필요한 정보인지 확인하시고 미사용필드들은 전부제거하면되고,
dto관리측면에서 복잡하다는것을 의미하셨다면 dto assembler등 관리하는 방법들이 있으니 참고해보셔요.
답변이 되었을까요?

bugoverdose replied:
"클래스가 최대 2개의 필드만을 지녀야 한다"는 조건이 습관화되어서 dto 내에 다른 dto들이 과하게 중첩된 구조가 생긴 것 같아요. 미사용 필드나 불필요하다고 느껴지는 클래스들도 제거해봤습니다.

JunHoPark93 commented on src/test/java/study/HashingTest.java:
```diff
@@ -0,0 +1,49 @@
+        factory = SecretKeyFactory.getInstance("PBKDF2WithHmacSHA1");
+    }
+
+    @Test
+    void 동일한_입력값과_salt에_대한_해쉬값은_동일() throws InvalidKeySpecException {
+        KeySpec spec = new PBEKeySpec(PASSWORD.toCharArray(), SALT, 65536, 128);
+        byte[] hashByte1 = factory.generateSecret(spec).getEncoded();
+
+        KeySpec spec2 = new PBEKeySpec(PASSWORD.toCharArray(), SALT, 65536, 128);
+        byte[] hashByte2 = factory.generateSecret(spec2).getEncoded();
+
+        assertThat(hashByte1).isEqualTo(hashByte2);
+        // System.out.println(Base64.getEncoder().encodeToString(hashByte1));
```
동일한 입력의 다른 salt값에 대한 테스트도 있으면 좋을것같아요~

JunHoPark93 reviewed (changes requested):
안녕하세요 정! 
2단계도 벌써 구현해주셨네요~
질문주신 부분들과 몇가지 코멘트 남겨두었는데요, 확인 후 의견 및 반영부탁드려요!
궁금하신것 편하게 남겨주셔요~

bugoverdose committed:
chore(db): change db port to 13306 to prevent collision

- add dev env configuration instructions on README

bugoverdose committed:
refactor(service): move encryption logic to controller layer

bugoverdose committed:
test(service): add test cases on deleteFinishedGame

bugoverdose committed:
refactor(controller): remove SearchController

- /search to /game/search

bugoverdose committed:
feat(config): implement AuthArgumentResolver

- read json data at request body and parse with BasicJsonParser
- send encrypted auth data to parameters at controller

bugoverdose committed:
test(study): add tests on hashing with salt

bugoverdose committed:
docs(README): add multiplayer feature list

bugoverdose committed:
feat(web): implement entering chess room logic with password

- set dummy data as cookie on response

bugoverdose committed:
feat(util): implement HashUtils for hashing

bugoverdose committed:
refactor(auth): separate hashing logic to utils

- remove unnecessary static factory method

bugoverdose committed:
feat(utils): implement CookieUtils for generating and validating cookies

bugoverdose committed:
feat(controller): set encrypted cookie on browser on creating new game

bugoverdose committed:
refactor(chess): remove unnecessary CreatedGameDto

- return single int value instead of json format

bugoverdose committed:
feat(utils): separate cookies via path(game/:id)

bugoverdose committed:
refactor(dto): cleanse dtos

- remove unnecessary GameStateDto
- change dto transform logic at Game and use cases

bugoverdose committed:
refactor(domain): cleanse Game implementations

- remove verbose template method pattern
- override all equals, hashCode, toString methods at Running

bugoverdose committed:
feat(utils): implement GameStateDisplayUtil

- to separate display logic from domain

bugoverdose committed:
refactor(domain): cleanse game package

- remove display logic from GameState enum

bugoverdose committed:
feat(dto): implement static factory methods at dtos

- cleanse dto transformation logic at service

bugoverdose committed:
test(dao): add missing test case

- refactor gameDao tests

bugoverdose committed:
chore(chess): comment out step3 features

bugoverdose committed:
refactor(dto): cleanse dto structure

- delete unnecessary BoardViewDto
- add missing access modifiers

bugoverdose committed:
refactor(domain): cleanse game state logic

- add missing access modifiers
- remove template method pattern on getter

bugoverdose committed:
feat(dao): implement StatementExecutor for exception handling

bugoverdose committed:
refactor(dao): replace legacy code with StatementExecutor

bugoverdose committed:
refactor(chess): remove all step3 features

- rollback README

bugoverdose commented:
미션 추가 요구사항 관련 커밋들을 최대한 지워보려고 했는데 여전히 커밋이 좀 지저분하네요. Cookie 관련 커밋 내역은 전부 무시해주시기 바랍니다.

1. DTO, DAO, Controller 리팩토링 위주로 수정하였고, 
2. 비밀번호 암호화 작업을 ArgumentResolver에서 수행하여 컨트롤러에 넘겨주는 방식으로 수정해보았습니다. 
3. 도커 관련하여 제가 개발환경을 셋업한 방식을 README에 기록해놓았고, 혹시 몰라서 포트번호를 3306에서 13306으로 수정하였습니다. 

혹시 여전히 실행시 문제 생기실까봐 정상 동작했을 때의 방식을 영상으로 남겨놓았습니다.

[![참고영상](https://user-images.githubusercontent.com/73531614/165442011-5d3f67a3-c5ad-428e-9fac-b5a50c82c313.png)](https://user-images.githubusercontent.com/73531614/165441224-8fc176c3-c53b-4a86-bfb5-c0340ded43e5.mov)

bugoverdose committed:
fix(MoveRoute): implement MoveRouteDto for mapping json

bugoverdose committed:
feat(web): change exception message on search fail

JunHoPark93 commented:
<img width="1035" alt="스크린샷 2022-04-28 오후 10 02 13" src="https://user-images.githubusercontent.com/20608121/165758081-1ed85acc-52bd-42e2-82bc-dfdafbd3473c.png">
잘 되는것 확인했습니다~

JunHoPark93 commented:
잘해주셨고 시간이 좀 남으셨는데 https://spring.io/projects/spring-framework 공식 문서도 한번 보시면 나중에 도움될거에요.

JunHoPark93 reviewed (approved):
안녕하세요 정~
리뷰 잘 반영해주셨어요.
여러 가지 시도를 해보고 적용도 잘해보셨네요~
일찍 마무리가 되었으니 추가로 더 보면 좋을것 남겨둘게요.
고생많으셨고 다음미션도 화이팅에요~
